### PR TITLE
Update PlatformLib to `v0.2.10` (resubmit for CI)

### PR DESCRIPTION
**Pull request by gavanderhoorn**
(This is a replacement for #89 to get CI working (secrets don't work for workflows triggered by events from forks))

As per subject.

Only build-tested (but we'll see what CI thinks of it), although we have tested ~~this~~ *similar* versions of the platform lib as part of #49.

0ec05f5ea8db598e4a855da72dda4a659fec165a adds the changes necessary to use the new version of `Ros_GetMacAddress(..)` (which now requires an argument to indicate for *which* interface the MAC should be retrieved).

e202c7f63076c0026bbe6ecfb68635dcd6f634ec addresses #60, at least for `SUBCODE_FAIL_MP_NICDATA` (it was used 4 times in total).

### CODE REVIEW

File: src/CommunicationExecutor.c [hunk 1]
```diff
@@ -61,11 +61,29 @@ void Ros_Communication_ConnectToAgent()
...
-    motoRosAssert_withMsg(
-        Ros_GetMacAddress(macId) == OK,
-        SUBCODE_FAIL_MP_NICDATA,
-        "Must enable ETHERNET function");
+    STATUS status = Ros_GetMacAddress(ROS_USER_LAN1, macId);
+    if (status != OK)
+    {
+        Ros_Debug_BroadcastMsg("%s: Ros_GetMacAddress: iface: %d; error: %d",
+            __func__, ROS_USER_LAN1, status);
+        motoRosAssert_withMsg(FALSE, SUBCODE_FAIL_MP_NICDATA_INIT0,
```
Comment (ted-miller) [inline src/CommunicationExecutor.c:72]: On YRC, this won't allow the code to get to the point where it checks LAN2

File: src/ConfigFile.c [hunk 2]
```diff
@@ -132,10 +132,29 @@ void Ros_ConfigFile_SetAllDefaultValues()
...
-        "Must enable ETHERNET function");
+
+    //TODO(gavanderhoorn): get MAC only from interface which is used for
+    //ROS traffic (https://github.com/Yaskawa-Global/motoros2/issues/57)
+    STATUS status = Ros_GetMacAddress(ROS_USER_LAN1, macId);
+    if (status != OK)
+    {
+        Ros_Debug_BroadcastMsg("%s: Ros_GetMacAddress: iface: %d; error: %d",
+            __func__, ROS_USER_LAN1, status);
+        motoRosAssert_withMsg(FALSE, SUBCODE_CONFIGURATION_FAIL_MP_NICDATA0,
```
Comment (ted-miller) [inline src/ConfigFile.c:143]: Same issue here

**Review by ted-miller — CHANGES_REQUESTED** [review]

File: src/CommunicationExecutor.c (hunk 1 above)
Comment (gavanderhoorn) [reply src/CommunicationExecutor.c:72]: I know. But do we want to just try any interface which works, or should we actually just raise an alarm if we can't retrieve the MAC for the interface we're asking for?

I would probably suggest to do the first, as the previous behaviour could lead to situations where MotoROS2 ends up trying to use the 3rd interface while the 2nd is actually the one with the network cable (ie: for some reason, 2nd interface MAC can't be read, 3rd interface succeeds -> we just continue with that interface).

Comment (ted-miller) [reply src/CommunicationExecutor.c:72]: > I would probably suggest to do the first, as the previous behaviour could lead to situations where MotoROS2 ends up trying to use the 3rd interface while the 2nd is actually the one with the network cable (ie: for some reason, 2nd interface MAC can't be read, 3rd interface succeeds -> we just continue with that interface).

I don't understand this comment. I'm not sure which of the two you're suggesting.

Personally, I'd just go with any interface that works.

Also, if we do maintain this assertion, then the check for the second MAC will never logically happen. Either status is bad and there's an alarm, or status is good and the second one never gets checked.

Comment (gavanderhoorn) [reply src/CommunicationExecutor.c:72]: Failure to read the MAC is not necessarily related to whether or not an interface is `UP` and/or `RUNNING`.

We're already tracking the case where with two interfaces connected, MotoROS2 always uses the first interface (ie: `ROS_USER_LAN1`) to get the MAC (#57) and to broadcast the debug logs (#58).

I've learned that failing early (and loudly) is better than trying to continue with whatever fallback does work.

Hence the question/comment.

> Also, if we do maintain this assertion, then the check for the second MAC will never logically happen. Either status is bad and there's an alarm, or status is good and the second one never gets checked.

yes, agreed.

0ec05f5ea8db598e4a855da72dda4a659fec165a was essentially just the old pattern but adapted to:

 - the changed `Ros_GetMacAddress(..)` implementation, and
 - no longer reuse alarm codes

Comment (gavanderhoorn) [reply src/CommunicationExecutor.c:72]: @ted-miller: would implementing the previous behaviour (ie: just try interfaces until one works) unblock this PR?

If so, I'll make that change and push it here.

We can update/correct later if desired.

**Review by ted-miller — APPROVED** [review]

**Review by ted-miller — APPROVED** [review]

**Pull request merged**